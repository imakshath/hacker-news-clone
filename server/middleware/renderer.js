import React from 'react'
import ReactDOMServer from 'react-dom/server'
import { Provider as ReduxProvider } from 'react-redux';
// import our main App component
import App from '../../src/App';

const path = require("path");
const fs = require("fs");

export default (store) => (req, res, next) => {
    // point to the html file created by CRA's build tool
    const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');

    fs.readFile(filePath, 'utf8', (err, htmlData) => {
        if (err) {
            return res.status(404).end()
        }

        const html = ReactDOMServer.renderToString(
            <ReduxProvider store={store}>
                <App />
            </ReduxProvider>
        );

        // inject the rendered app into our html and send it
        return res.send(
            htmlData.replace(
                '<div id="root"></div>',
                `<div id="root">${html}</div>`
            )
        );
    });
}