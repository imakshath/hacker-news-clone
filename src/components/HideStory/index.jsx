import React from 'react';

function HideStory({ onHideStory, storyId }){
    const handleHideStory = React.useCallback(() => {
        onHideStory(storyId);
    }, [onHideStory, storyId]);
    return (
        <span data-testid={`hide-story-${storyId}`} className="hide-action-wrapper ml-5" onClick={handleHideStory}>[ hide ]</span>
    )
};

export default React.memo(HideStory);