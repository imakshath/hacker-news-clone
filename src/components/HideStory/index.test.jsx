import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HideStory from '.';

afterEach(cleanup);

describe('<HideStory /> component', () => {
    let wrapper;
    const props = {
        storyId: "123",
        onHideStory: jest.fn()
    };
    
    beforeEach(() => {
        wrapper = render(
            <HideStory {...props} />
        );
    });

    test('should render without crash', () => {
        expect(wrapper.container).toBeDefined();
    });

    test('should get by testid', () => {
        expect(wrapper.getByTestId('hide-story-123')).toBeDefined();
        expect(wrapper.getByText('[ hide ]')).toBeDefined();
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<HideStory />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})