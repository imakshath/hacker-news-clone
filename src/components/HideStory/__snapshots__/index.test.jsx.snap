// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HideStory /> component should match snapshot 1`] = `
<span
  className="hide-action-wrapper ml-5"
  data-testid="hide-story-undefined"
  onClick={[Function]}
>
  [ hide ]
</span>
`;
